### DISCUSSION TITLE
Explicit Interface Method Implementation is a design best practice

### DISCUSSION THREAD

**Discussion by arialdomartini** [arialdomartini/arialdomartini.github.io#5] [General]
Comments to the article [Explicit Interface Method Implementation is a design best practice](https://arialdomartini.github.io/explicit-implementation)

**Comment by arphox** [comment]
Thanks for this intriguing article!

I like your reasoning and I am thinking whether I should default to explicitly implementing interfaces from now on.

I'd like to ask you to actively find some disadvantages with using EIMIs and add them to your great article.

For example, see this:  
https://stackoverflow.com/a/4949060/4215389

**Reply by arialdomartini** [reply]
Thank you @arphox.
I like the suggestion to deliberately think to the counter arguments. I promise I will do my best. I'm just sorry if at the moment, I don't find any.

By coincidence, right few hours ago @dsyme closed as declined [a 7 year old proposal on F#](https://github.com/fsharp/fslang-suggestions/issues/195), exactly on supporting implicit interface implementation. 

I got an email notification because I happened to be the last one to post a comment there, almost 3 years ago. Very curiously, that was around this very topic: how I could not find a reason why F# should amend the right decision to only support explicit implementation. Funny.

I'll take the goal of finding some counter arguments as a challenge, promised. Being madly in love with F# won't help, though :)
